framework/core/js/src/common/Application.tsx

```diff
@@ -356,9 +358,61 @@ export default class Application {
 
     document.body.classList.add('ontouchstart' in window ? 'touch' : 'no-touch');
 
+    this.initColorScheme();
+
     liveHumanTimes();
   }
 
+  private initColorScheme(forumDefault: string | null = null): void {
+    forumDefault ??= document.documentElement.getAttribute('data-theme') ?? 'auto';
+    this.allowUserColorScheme = forumDefault === 'auto';
+    const userConfiguredPreference = this.session.user?.preferences()?.colorScheme;
+
+    let scheme;
+
+    if (this.allowUserColorScheme) {
+      scheme = userConfiguredPreference;
+    }
+
+    scheme ||= forumDefault;
+
+    this.setColorScheme(scheme);
+
+    // Listen for browser color scheme changes and update the theme accordingly
+    if (this.allowUserColorScheme) {
+      this.watchSystemColorSchemePreference(() => {
+        this.initColorScheme(forumDefault);
+      });
+    }
+  }
+
+  getSystemColorSchemePreference(): string {
```

Comment:
Why not type this as a string union?

Reply:
changed to `ColorScheme | string` but kept string because extensions may add their own color schemes and extend these methods.